frontend/config-assistant/src/components/gui-editor/ConfigTreeNodeResolver.ts

```diff
@@ -45,11 +45,20 @@ export class ConfigTreeNodeResolver {
       type: nodeType,
       key: pathToString(absolutePath),
       children: [],
-      leaf: this.isLeaf(schema, depth),
+      leaf: this.isLeaf(schema, absolutePath, depth),
     };
   }
 
-  private isLeaf(schema: JsonSchema, depth: number): boolean {
+  private isLeaf(schema: JsonSchema, absolutePath: Path, depth: number): boolean {
+    const dependsOnUserSelection = schema.anyOf.length > 0 || schema.oneOf.length > 0;
+    if (dependsOnUserSelection) {
```

Comment:
for now its fine but actually we would need two different components: One for oneOf (single selection), one for multiple selection (anyOf) -> https://primevue.org/multiselect/